src/hotspot/share/opto/macro.cpp

```diff
@@ -2445,6 +2445,7 @@ void PhaseMacroExpand::eliminate_macro_nodes() {
 //------------------------------expand_macro_nodes----------------------
 //  Returns true if a failure occurred.
 bool PhaseMacroExpand::expand_macro_nodes() {
+  C->dont_allow_macro_nodes();
```

Comment:
Leave a comment here:

`// Do not allow new macro nodes once we started to expand`